### PR TITLE
fix: Fix Add Command PE-D issues

### LINKED ISSUE #159: [PE-D][Tester E] Remarks changes from null to empty upon restart of application
-------------
Labels: `severity.VeryLow` `type.FunctionalityBug`
original: itsmejr257/ped#7

### LINKED ISSUE #182: [PE-D][Tester C] Lack of /help page
Would be good to include a help page so users do not need to keep referring to the userguide on a different tab.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: cayennegoh/ped#9

### LINKED ISSUE #184: [PE-D][Tester F] using an example from the user guide is causing an error
-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: nikhil-2101/ped#1

### LINKED ISSUE #191: [PE-D][Tester F] empty imput is allowed when adding
-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: nikhil-2101/ped#5

### PR DESCRIPTION
**Pull request by nickczh**
# Checklist for feature implementation and modification
Closes #159, Closes #182, Closes #184, Closes #191
- [x] Added relevant milestone (under milestone tab on the right)
- Implementation
  - [x] Sources (if any) credited under both relevant section(s) and Developer Guide (Acknowledgement)
  - [x] JUnit Test(s)
- Documentation
  - [x] JavaDoc for classes and methods
  - Developer Guide
    - [ ] Design (Can add as a todo statement for later on)
    - [ ] Implementation (Can add as a todo statement for later on)
    - [ ] Relevant User Story present
    - [ ] Non-functional requirement(s) that the feature addresses (if any)
    - [ ] Glossary of the (technical) term(s) used (if any)
    - [ ] Specialised instructions for manual testing (if any)
  - User Guide
    - [x] Description and usage of feature (if any)
  - Contributions (under `/docs/team/*.md`)
    - [x] Code/Feature/Enhancement contribution
    - [x] Developer Guide contribution
    - [x] User Guide contribution
    - [ ] Misc and team-based admin tasks (if any)

# Checklist for non-implementation related PR
Closes #<Issue_Number>
- [ ] Added relevant milestone (under milestone tab on the right)
- Documentation (Contributions)
  - [ ] Developer Guide contributions (if any)
  - [ ] User Guide contributions (if any)
  - [ ] Review contributions (if any)
  - [ ] Miscellaneous and team-based admin tasks (if any)

# Additional Information

Nil

### CODE REVIEW

File: src/main/java/meditracker/command/AddCommand.java [hunk 1]
```diff
@@ -56,6 +56,8 @@ public class AddCommand extends Command {
     private double medicationDosageMorning = 0.0;
     private double medicationDosageAfternoon = 0.0;
     private double medicationDosageEvening = 0.0;
+    private int repeat;
+    private String remarks = "Nil";
```
Comment (e0958902) [inline src/main/java/meditracker/command/AddCommand.java:60]: We may want to update the User Guide to state that empty remarks will be stated as "Nil", especially for users who is easily baffled. 
image

File: src/main/java/meditracker/command/HelpCommand.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+/**
+ * Represents a command to display help information for all available commands in MediTracker.
+ *
+ * This command lists all available commands along with a brief description of each.
+ * It extends the Command class and overrides the execute method to perform the action.
+ */
+public class HelpCommand extends Command {
+
+    /**
+     * Executes the help command by displaying a list of available commands and descriptions.
```
Comment (e0958902) [inline src/main/java/meditracker/command/HelpCommand.java:14]: You may want to consider changing "by displaying" to "to display".

**Review by e0958902 — COMMENTED** [review]
Due to the interest of time, here are some comments even though this pull request is still draft mode,

File: src/main/java/meditracker/command/HelpCommand.java (hunk 2 above)
Comment (nickczh) [reply src/main/java/meditracker/command/HelpCommand.java]: Resolved in https://github.com/AY2324S2-CS2113T-T09-1/tp/pull/210/commits/827dd72b58a9ea6835e368e7a8aa42dfb58b14ec. Thanks.

File: src/main/java/meditracker/command/AddCommand.java (hunk 1 above)
Comment (nickczh) [reply src/main/java/meditracker/command/AddCommand.java:60]: Resolved in https://github.com/AY2324S2-CS2113T-T09-1/tp/pull/210/commits/c233429ef3d492b0911b3cc875afcabe87a79322. Thank you!

File: src/main/java/meditracker/command/HelpCommand.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        System.out.print(System.lineSeparator());
+        System.out.println("For more details about each command, simply type in the command name.");
+    }
+
+    /**
+     * Retrieves and prints the names and descriptions of all available commands, except for the UNKNOWN command.
+     * Iterates through the CommandName enum values, skipping the UNKNOWN command,
+     * and prints each command's name and its description to the console.
+     */
+    public void getCommandNamesAndDescription() {
```
Comment (wenenhoe) [inline src/main/java/meditracker/command/HelpCommand.java:33]: Can consider storing the string to print, as a static variable of this class, so it doesn't need to process again and generate the help message each time this method is invoked

File: src/main/java/meditracker/command/CommandParser.java
Comment (wenenhoe) [inline src/main/java/meditracker/command/CommandParser.java:1]: #184 seems to still persist
Might want to add a `.strip()` call in between the `fullCommand.split(" ", 2)` on L29 to resolve this

```suggestion
        String[] commands = fullCommand.strip().split(" ", 2);
```

**Review by wenenhoe — CHANGES_REQUESTED** [review]
Good job on writing on tests for `AddCommand` and thanks for helping to squash those bug reports from PE-D. Left a few suggestions for your consideration

**Review by wenenhoe — APPROVED** [review]
LGTM! Thanks for rectifying the changes

File: src/main/java/meditracker/command/HelpCommand.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+     * It prints a header message followed by each command's name and description.
+     */
+    @Override
+    public void execute() {
+        System.out.print(System.lineSeparator());
+        Ui.showLine();
+        System.out.println("Here are the commands you can use with MediTracker:" +
+                System.lineSeparator());
+        getCommandNamesAndDescription();
+        System.out.print(System.lineSeparator());
```
[the remark this answers is not in the document]
Comment (nickczh) [reply src/main/java/meditracker/command/HelpCommand.java]: Resolved in https://github.com/AY2324S2-CS2113T-T09-1/tp/pull/210/commits/4b6e81c1e1e59fcdd257dc1eee063843334c0d7c. Thank you.

File: src/main/java/meditracker/command/HelpCommand.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+     * Retrieves and prints the names and descriptions of all available commands, except for the UNKNOWN command.
+     * Iterates through the CommandName enum values, skipping the UNKNOWN command,
+     * and prints each command's name and its description to the console.
+     */
+    public void getCommandNamesAndDescription() {
+        for (CommandName commandName : CommandName.values()) {
+            if (commandName.equals(CommandName.UNKNOWN)) {
+                continue;
+            }
+            System.out.println(commandName.value + ": " + commandName.description);
```
[the remark this answers is not in the document]
Comment (nickczh) [reply src/main/java/meditracker/command/HelpCommand.java]: Resolved in https://github.com/AY2324S2-CS2113T-T09-1/tp/pull/210/commits/546503795e3aed43243b3df1802393145e832f87. Thanks!

File: src/main/java/meditracker/command/AddCommand.java [hunk 6]
```diff
@@ -94,31 +97,43 @@ public void execute() {
...
+        String repeatArg = parsedArguments.get(ArgumentName.REPEAT);
 
-        int repeat;
         try {
+            // Validate and parse input
+            sanitiseInput(medicationName);
             repeat = getRepeat();
             parseStringToValues(medicationQuantityArg, medicationDosageMorningArg,
-                    medicationDosageAfternoonArg, medicationDosageEveningArg);
+                    medicationDosageAfternoonArg, medicationDosageEveningArg, repeatArg, remarksArg);
```
[the remark this answers is not in the document]
Comment (nickczh) [reply src/main/java/meditracker/command/AddCommand.java]: Resolved in https://github.com/AY2324S2-CS2113T-T09-1/tp/pull/210/commits/fc0fbc867d3262c1751f36b121428ce22124f223. Thanks.

File: src/main/java/meditracker/command/CommandParser.java
Comment (nickczh) [reply src/main/java/meditracker/command/CommandParser.java:1]: Resolved in https://github.com/AY2324S2-CS2113T-T09-1/tp/pull/210/commits/8513dace26abcf367d4b9031da7ed948e5d87728. Thanks!

**Pull request merged**